core/src/impl/Kokkos_Stacktrace.cpp

```diff
@@ -0,0 +1,265 @@
+#include "Kokkos_Stacktrace.hpp"
+
+// backtrace() function for retrieving the stacktrace
+#include <execinfo.h>
```

Comment:
See comments on previous PR.  I've been trying to figure out how to add a TPL to Kokkos.  I can do it just fine with TriBITS, but Kokkos has some insane Makefile-driven thing where you have to add a bunch of code in different places.  I can't quite get the TPL's macro to show up in `KokkosCore_config.h` yet.  If you have some insight, please let me know.

Reply:
Uhm what TPL are you adding here??

Reply:
`execinfo.h` is part of GNU Library (man page - http://man7.org/linux/man-pages/man3/backtrace.3.html)

Reply:
@nmhamster We've only tried this with GCC and Clang.  I'm pretty sure Intel works since they strive for GCC compatibility, but what about the XL or other compilers?  The `backtrace*` functions are not in the POSIX standard.  Also, this won't work on Windows as far as I know.